Common/ExecutionModel/vtkThreadedImageAlgorithm.cxx

```diff
@@ -25,12 +25,105 @@
 #include "vtkPointData.h"
 #include "vtkStreamingDemandDrivenPipeline.h"
 
+#include "vtkExtentTranslator.h"
+
+
+
+//---- option 2
+
+     #include <tbb/parallel_for.h>
+#include <tbb/blocked_range.h>
+#include <tbb/blocked_range3d.h>
+#include <tbb/blocked_range2d.h>
+#include <tbb/task_scheduler_init.h>
+//-- option 1
+#include "vtkSMPTools.h"
+
+
+
+
+
+
+struct vtkImageThreadStruct
+{
+vtkThreadedImageAlgorithm *Filter;
+vtkInformation *Request;
+vtkInformationVector **InputsInfo;
+vtkInformationVector *OutputsInfo;
+vtkImageData   ***Inputs;
+vtkImageData   **Outputs;
+};
+
+class apply_transform{
+  
+public:
+  vtkImageThreadStruct* str;
+  apply_transform(vtkImageThreadStruct* abc)
+  {
+    str = abc;
+  }
+  apply_transform()
+  {
+   
+  }
+  void operator()(const tbb::blocked_range2d<vtkIdType>& r) const {
+
+    int extent[6];
+    extent[0] = r.cols().begin();
+    extent[1] = r.cols().end();
+    extent[2] = r.rows().begin();
+    extent[3] = r.rows().end();
+    extent[4] = 0;
+    extent[5] = 0;    
+    //r.cols
+    str->Filter->ThreadedRequestData(str->Request,
+      str->InputsInfo, str->OutputsInfo,
+      str->Inputs, str->Outputs,
+      extent, 1);
+  }
+};
+
+
+
+class vtkThreadedImageAlgorithmFunctor
+{
+  vtkImageThreadStruct *str;
+  int * extents;
```

Comment:
It's much safer to declare this as an array instead of a pointer:

    int Extents[6];

A good rule is this: a object's member variable should not be a pointer, unless the memory that is being pointed to is managed by the object.  Ideally, the "str" member shouldn't be a pointer, either.